feat: Draft goals

todaywasawesome opened the pull request:
Closes #3, this draft is considered a work in progress. Next steps are
[x] Committee Review
[x] Solicit Maintainer Feedback
[x] Merge

When providing feedback, please select the specific lines you'd like to address and even feel free to make edits using Github's suggestion tool. We will discuss these in the next conformance committee meeting. 

For context, consider reviewing other successful CNCF conformance programs:
* https://www.cncf.io/certification/software-conformance/
* https://github.com/cncf/crossplane-conformance
* https://github.com/cncf/prometheus-conformance

Vendors are already building forks of Argo meant to compete with the open-source offering and community. This underlines the value of building this kind of program and encouraging more community cohesion. 

Co-authored-by: Jann Fischer <jfischer@redhat.com>
Co-authored-by: Henrik Blixt <henrik_blixt@intuit.com>
Co-authored-by: Hong Wang <wanghong230@gmail.com>

todaywasawesome committed:
Draft goals

Signed-off-by: Dan Garfield <dan@todaywasawesome.com>

todaywasawesome commented on README.md:
```diff
@@ -1,2 +1,19 @@
+Software conformance ensures predictability and helps protect the Argo brand. 
+
+# Goals
+* Ensure a consistent and reliable user experience
+* Protect the Argo brand
```
```suggestion
* Ensure a consistent and reliable user experience and API compatibility 
* Protect the Argo brand
```

From Henrik

todaywasawesome committed:
Update README.md

Make sure API compatibility is called out because it is key.

Signed-off-by: Dan Garfield <dan@todaywasawesome.com>

todaywasawesome commented on README.md:
```diff
@@ -1,2 +1,19 @@
+# Goals
+* Ensure a consistent and reliable user experience
+* Protect the Argo brand
+* Enable contributions from commercial partners
+* To grow and strengthen the Argo community
```
```suggestion
* Ensure contributions from commercial partners
* To grow and strengthen the Argo community
```

We will likely require active maintainers in order to meet conformance. It's not enough to encourage contributions, they should be required.

todaywasawesome committed:
Update README.md

Signed-off-by: Dan Garfield <dan@todaywasawesome.com>

todaywasawesome commented on README.md:
```diff
@@ -1,2 +1,19 @@
+
+# Benefits of Certification
+In addition to giving users confidence and protecting the Argo community certified offerings may
+* Advertise compliance and certification
```
```suggestion
* Advertise compliance and certification
* Use Argo logo and branding in marketing materials
```

Make sure that brand usage is explicitly stated.

todaywasawesome committed:
Update README.md

Signed-off-by: Dan Garfield <dan@todaywasawesome.com>

todaywasawesome commented on README.md:
```diff
@@ -1,2 +1,19 @@
+# Benefits of Certification
+In addition to giving users confidence and protecting the Argo community certified offerings may
+* Advertise compliance and certification
+* Preference in community and marketing events etc.
```
```suggestion
* Exposure to community
```

Simplify language and avoid "preference".

todaywasawesome committed:
Update README.md

Signed-off-by: Dan Garfield <dan@todaywasawesome.com>

todaywasawesome commented on README.md:
```diff
@@ -1,2 +1,20 @@
+Software conformance ensures predictability and helps protect the Argo brand. 
+
+# Goals
+* Ensure a consistent and reliable user experience and API compatibility 
+* Protect the Argo brand
+* Ensure contributions from commercial partners
+* To grow and strengthen the Argo community
```
```suggestion
* Guarantee a consistent and reliable user experience and API compatibility 
* Protect the Argo brand
* Ensure contributions from commercial partners
* Grow and strengthen the Argo community
```
Make lines consistent in form and don't reuse "ensure" too much.

todaywasawesome committed:
Update README.md

Signed-off-by: Dan Garfield <dan@todaywasawesome.com>

wanghong230 commented on README.md:
```diff
@@ -1,2 +1,21 @@
+* Guarantee a consistent and predictable user experience
+* Guarantee interoperability, including API compatibility
+* Guarantee conformant products are up-to-date with latest Argo releases 
+* Promote contributions from commercial partners
```
What is the reason to change to Promote?

edlee2121 replied:
Is Encourage better?

todaywasawesome replied:
@edlee2121 for any additional changes please use the code suggest feature in a code review. Which will allow people to discuss the change before merging. 

We actually started with "Encourage" and the group felt that it wasn't strong enough as a goal and that we should basically require contributions to be coming back in some way. It's likely this will take the form of certified offerings to have a maintainer in the Argoproj. 

Given that I think we should go back to "Ensure".

todaywasawesome replied:
```suggestion
* Ensure contributions from commercial partners
```

wanghong230 replied:
I like "Ensure" as the goal to really benefit the community in a measurable way. Thanks.

edlee2121 replied:
Sure not sure how to use that feature. Let me revert my commit and resubmit it.

edlee2121 commented on README.md:
```diff
@@ -1,2 +1,20 @@
+
 # Argo Conformance Program
-A standard to ensure interoperability, protect users, and prevent surprises when using commercial offerings with Argo in the name. Currently in draft and non-binding. 
+The Argo community continues to grow dramatically – adding new users, new contributors, and new capabilities. The Argo Conformance Program creates incentives and requriements designed to help the ecosystem grow in a sustainable and user-friendly way. As a diverse set of tools grows around Argo, users need to know when they can they can trust these tools to work together and in a predictable way.
```
```suggestion
+The Argo community continues to grow dramatically – adding new users, new contributors, and new capabilities. The Argo Conformance Program creates incentives and requirements designed to help the ecosystem grow in a sustainable and user-friendly way. As a diverse set of tools grows around Argo, users can trust that products certified as compliant will interoperate with each other and be up-to-date with the latest open source releases. This ensures a consistent user experience between products and flexibility to choose between vendors.
```

todaywasawesome replied:
LGTM 👍

edlee2121 commented on README.md:
```diff
@@ -1,2 +1,20 @@
+Software conformance ensures predictability and helps protect the Argo brand. 
+
+# Goals
+* Guarantee a consistent and reliable user experience and API compatibility 
+* Protect the Argo brand
+* Ensure contributions from commercial partners
+* Grow and strengthen the Argo community
```
```suggestion
+* Guarantee a consistent and predictable user experience
+* Guarantee interoperability, including API compatibility
+* Guarantee conformant products are up-to-date with latest Argo releases 
+* Ensure contributions from commercial partners
 * Grow and strengthen the Argo community
+* Protect the Argo brand

```

todaywasawesome replied:
Good additions and the wording is solid. 👍

jannfis reviewed (approved):
LGTM

todaywasawesome committed:
Fix typos and reword consistency language to be more clear.

Signed-off-by: Dan Garfield <dan@codefresh.io>

Co-authored-by: Ed Lee <edlee2121@users.noreply.github.com>

todaywasawesome committed:
Improve goals language, add keeping conformance up-to-date

Signed-off-by: Dan Garfield <dan@codefresh.io>

Co-authored-by: Ed Lee <edlee2121@users.noreply.github.com>

edlee2121 commented on README.md:
```diff
@@ -1,2 +1,23 @@
+In addition to giving users confidence and protecting the Argo community certified offerings may
+* Advertise compliance and certification
+* Use Argo logo and branding in marketing materials
+* Exposure to community
```
```suggestion
* Exposure to community

# Benefits to Community
* Promote interoperability and compatibility between Argo products
* Discourage forking of the Argo code base
* Encourage commercial interests to work together to grow the Argo ecosystem rather than purely on competition with each other
```

todaywasawesome replied:
I like it!

todaywasawesome committed:
Add benefits to the community

Co-authored-by: Ed Lee <edlee2121@users.noreply.github.com>

hblixt commented on README.md:
```diff
@@ -1,2 +1,28 @@
++* Guarantee a consistent and predictable user experience
++* Guarantee interoperability, including API compatibility
++* Guarantee conformant products are up-to-date with latest Argo releases 
++* Ensure contributions from commercial partners
```
Do we want to use something softer, like "engagement" over contributions? As a goal, engagement feels better, then we can specify in the requirements exactly what it means

todaywasawesome replied:
I'm good with that @hblixt. @edlee2121 made a good point when I talked to him today about a race for people to get maintainers in. I think we can work it out in the requirements but I'm fine with either word.

hblixt replied:
I don't really feel strongly either way. It just felt a little more appropriate for a goal to use engagement over contributions, which seemed a bit more of a requirement.

todaywasawesome replied:
Debated this in committee with @alexec @wanghong230 and @hblixt decided to move forward with current language. May revisit if someone wants conformance without maintainers.

hblixt commented on README.md:
```diff
@@ -1,2 +1,28 @@
+# Benefits to Community
+* Promote interoperability and compatibility between Argo products
+* Discourage forking of the Argo code base
+* Encourage commercial interests to work together to grow the Argo ecosystem rather than purely on competition with each other
```
We should leave out the negative part of the statement (ie "... rather than purely on competition with each other"). 
Focus on the positive :)

todaywasawesome replied:
```suggestion
* Encourage commercial interests to work together to grow the Argo ecosystem
```

todaywasawesome replied:
Discussed in committee with @hblixt @alexec and @wanghong230 agreed to remove the last part as it's covered in the first part.

todaywasawesome committed:
Remove negative language

Discussed in committee with @hblixt @alexec and @wanghong230 agreed to remove the last part as it's covered in the first part.
